### ISSUE TITLE
Application Insights Telemetry failing to initialize

### ISSUE THREAD

**Issue by dsarfati** [dotnet/orleans#4391]
I'm trying to add the Application Insights Telemtry to our 2.0 app and im getting the following exception 
```
System.InvalidOperationException: Unable to resolve service for type 'System.String' while attempting to activate 'Orleans.TelemetryConsumers.AI.AITelemetryConsumer'.
```
Looking at the code it looks like the only constructor on the `AITelemetryConsumer` takes a string, causing the exception. Simply adding a DI compatible constructor should fix the issue. I'll submit a PR if this makes sense.

**Comment by xiazen** [comment]
yeah that seems like a bug in our side. `AITelemetryConsumer ` consumer should have its own options. So that user can configure the options to give a `instrumentationKey`, 

To work around this, there's two options 
-  inject a string singleton to DI as instrumentalKey, so when AITelemetryConsumer is constructed through `ActivatorUtilities.CreateInstance(serviceProvider, consumerType);`, ActivatorUtility would locate that string singleton and create AITelemetryConsumer. But this is risky since it assumes you can only have one string singleton in DI. so I'm not recommending this one. 
- make simple adjustment to AITelemetryConsumer to fix the issue, something like 
```
 public class AITelemetryConsumerOptions
    {
        public string InstrumentationKey { get; set; }
    }

public class AITelemetryConsumer : ITraceTelemetryConsumer, IEventTelemetryConsumer, IExceptionTelemetryConsumer,
        IDependencyTelemetryConsumer, IMetricTelemetryConsumer, IRequestTelemetryConsumer
    {
         public AITelemetryConsumer(IOptions<AITelemetryConsumerOptions> options)
        {
            _client = options.Value.InstrumentationKey != null
                ? new TelemetryClient(new Microsoft.ApplicationInsights.Extensibility.TelemetryConfiguration { InstrumentationKey = options.Value.InstrumentationKey })
                : new TelemetryClient();
        }
   }
```

And then you can configure it through IOptions. 
```
new SiloHostBuilder.Configure<AITelemetryConsumerOptions>(op => op.InstrumentalKey = "")
.Configure<TelemetryOptions>(op=>op.AddConsumer<AITelemetryConsumer>());
```
- write your own TelemetryConsumer for AI 

At the mean time, I will submit a pr to fix the bug, but the fix will be included in our next release. So if you don't want to wait, you can do either of the workaround.

**Comment by xiazen** [comment]
submitted a PR #4392 to fix this

**Comment by dsarfati** [comment]
@xiazen Thanks! I actually ended up using option #2 until the fix is released. Thanks again for the help.

**Comment by el2iot2** [comment]
I see this is merged, but there is no new `Microsoft.Orleans.OrleansTelemetryConsumers.AI` [package ](https://www.nuget.org/packages/Microsoft.Orleans.OrleansTelemetryConsumers.AI/) out with the updated bits (still 2.0.0 there)? Is that an oversight or pending for another reason?

**Comment by sergeybykov** [comment]
Looks like we simply missed it. I guess we can relatively easily release a 2.0.3 version of these packages. I'll check.

**Comment by sergeybykov** [comment]
I opened #4687 for that.
